frontend/appflowy_flutter/integration_test/mobile/document/toolbar_test.dart

```diff
@@ -0,0 +1,131 @@
+      await editorState.apply(transaction);
+      await tester.pumpAndSettle();
+
+      /// insert two lines of text
+      const strFirst = 'FirstLine',
+          strSecond = 'SecondLine',
+          link = 'google.com';
+      await editorState.insertTextAtCurrentSelection(strFirst);
+      await tester.pumpAndSettle();
+      await editorState.insertNewLine();
+      await tester.pumpAndSettle();
+      await editorState.insertTextAtCurrentSelection(strSecond);
+      await tester.pumpAndSettle();
+      final firstLine = find.text(strFirst, findRichText: true),
+          secondLine = find.text(strSecond, findRichText: true);
+      expect(firstLine, findsOneWidget);
+      expect(secondLine, findsOneWidget);
+
+      /// select the first line
+      await tester.longPress(firstLine);
+      await tester.pumpAndSettle();
+
+      /// find aa item and tap it
+      final aaItem = find.byWidgetPredicate(
+        (widget) =>
+            widget is AppFlowyMobileToolbarIconItem &&
+            widget.icon == FlowySvgs.m_toolbar_aa_m,
+      );
+      expect(aaItem, findsOneWidget);
+      await tester.tapButton(aaItem);
+
+      /// find link button and tap it
+      final linkButton = find.byWidgetPredicate(
+        (widget) =>
+            widget is MobileToolbarMenuItemWrapper &&
+            widget.icon == FlowySvgs.m_toolbar_link_m,
+      );
+      expect(linkButton, findsOneWidget);
+      await tester.tapButton(linkButton);
+      await tester.pumpAndSettle();
+
+      /// input the link
+      await tester.pumpAndSettle();
+      final linkField = find.byWidgetPredicate(
+        (w) =>
+            w is FlowyTextField &&
+            w.hintText == LocaleKeys.document_inlineLink_url_placeholder.tr(),
+      );
+      await tester.enterText(linkField, link);
+      await tester.pumpAndSettle();
+
+      /// complete inputting
+      await tester.tapButton(find.text(LocaleKeys.button_done.tr()));
+      await tester.pumpAndSettle();
+
+      /// do it again
+      /// select the second line
+      await tester.longPress(secondLine);
+      await tester.pumpAndSettle();
+      await tester.tapButton(aaItem);
+      await tester.pumpAndSettle();
+      await tester.tapButton(linkButton);
+      await tester.pumpAndSettle();
+      await tester.enterText(linkField, link);
+      await tester.pumpAndSettle();
+      await tester.tapButton(find.text(LocaleKeys.button_done.tr()));
+      await tester.pumpAndSettle();
+
+      final nodes = editorState.document.root.children;
+
+      Map commonJson(String insert) => {
+            "type": "paragraph",
+            "data": {
+              "delta": [
+                {
+                  "insert": insert,
+                  "attributes": {"href": link},
+                }
+              ],
+            },
+          };
+
+      expect(
+        nodes.first.toJson(),
+        commonJson(strFirst),
+      );
+      expect(
+        nodes[1].toJson(),
+        commonJson(strSecond),
+      );
```

Comment:
```dart
      final firstNode = editorState.getNodeAtPath([0]);
      expect(firstNode?.type, ParagraphBlockKeys.type);
      expect(firstNode?.delta!.toJson(), [
        {
          "insert": strFirst,
          "attributes": {"href": link},
        }
      ]);
```

You can use `getNodeAtPath` to get the node

Reply:
All changes are completed